touchforms/formplayer/static/formplayer/script/formnav-all.js

```diff
@@ -93,6 +95,19 @@ function xformAjaxAdapter (formSpec, sessionData, savedInstance, ajaxfunc, submi
           getForm(q).reconcile(resp["tree"]);
         }
       });
+
+      if(this.answerCallback !== undefined) {
+          this.answerCallback(this.session_id);
+      }
+  }
+
+  this.evaluateXPath = function(xpath, callback) {
+    this.ajaxfunc({'action': 'evaluate-xpath',
+                   'session-id': this.session_id,
+                   'xpath': xpath},
+      function (resp) {
+        callback(resp['output']);
```

Comment:
if we raise the error on malformed xpath we can choose too different code paths for a success function and an error function. right now it works fine because we're just spitting whatever output is into a div, but in the future i think the separation between handling an error and handling a successful xpath query could be useful